src/pymmcore_widgets/_mda/_mda_widget.py

```diff
@@ -486,13 +489,14 @@ def _update_total_time(self) -> None:
         total_time: float = 0.0
         _per_timepoints: dict[int, float] = {}
         t_per_tp_msg = ""
+        _uses_time = self._uses_time()
 
         for e in self.get_state():
             if e.exposure is None:
                 continue
 
             total_time = total_time + (e.exposure / 1000)
-            if self._uses_time():
+            if _uses_time:
```

Comment:
I line profiled this and this was taking ~75% of the time in this function.

Reply:
wow.  good find